### DISCUSSION TITLE
Is there any special reason to work with forward_hook instead of backward_hook to get the gradients?

### DISCUSSION THREAD

**Discussion by tldrafael** [jacobgil/pytorch-grad-cam#335] [General]
Hi, thanks for the amazing work!

I am trying to implement a plain version of the GradCAM by myself. However, I tried to use [register_module_full_backward_hook](https://pytorch.org/docs/stable/generated/torch.nn.modules.module.register_module_full_backward_hook.html) instead of `register_forward_hook`, I got different values for many elements of the gradient tensor, although around 70% of tensor is equal. 

https://github.com/jacobgil/pytorch-grad-cam/blob/60cf39d6eb0e344d6921e62d42399a7fdf7cde57/pytorch_grad_cam/activations_and_gradients.py#L17

I wonder if there is anything that I did not figure out. Is it expected they show different results?
